internal/html/scrape/schema/json-ld/recipe.go

```diff
@@ -55,6 +56,11 @@ func (rp *RecipeProcessor) GetRecipeNode(doc *goquery.Document) (map[string]any,
 	)
 
 	for _, doc := range jsonLdDocs {
+		// Some websites (e.g. AllRecipes.com) have their schema wrapped in a list
+		doc = strings.TrimSpace(doc)
+		doc = strings.TrimPrefix(doc, "[")
```

Comment:
I think we should add code that is more resilient. What if `doc` is an array with more than one child?

We could add a helper method that attempts to deserialise as an object and, should that fail, then try to deserialise as an array.

Something like this:

```go
func unmarshalJsonObjOrArray(data string) (map[string]any, error) {
	var m map[string]any
	if err := json.Unmarshal([]byte(data), &m); err == nil {
		return m, nil
	}

	var nodes []any
	if err := json.Unmarshal([]byte(data), & nodes); err != nil {
		return nil, fmt.Errorf("unmarshal json list failed: %w", err)
	}

	for _, node := range nodes {
		if m, ok := node.(map[string]any); ok {
			return m, nil
		}
	}

	return nil, fmt.Errorf("unable to unmarshal data")
}
```

Reply:
implemented :)